tests/functional/codegen/calling_convention/test_external_contract_calls.py

```diff
@@ -1130,19 +1148,21 @@ def test_invalid_contract_reference_declaration(assert_tx_failed, get_contract):
 def __init__():
     pass
 """
-    assert_tx_failed(lambda: get_contract(contract), exception=StructureException)
+    with tx_failed(exception=StructureException):
+        get_contract(contract)
 
 
-def test_invalid_contract_reference_call(assert_tx_failed, get_contract):
+def test_invalid_contract_reference_call(tx_failed, get_contract):
     contract = """
 @external
 def bar(arg1: address, arg2: int128) -> int128:
     return Foo(arg1).foo(arg2)
 """
-    assert_tx_failed(lambda: get_contract(contract), exception=UndeclaredDefinition)
+    with tx_failed(exception=UndeclaredDefinition):
```

Comment:
i think this can just be `with pytest.raises(UndeclaredDefinition): compile_code(...)`